snappy/services.go

```diff
@@ -229,62 +190,47 @@ func removePackageServices(s *snap.Info, inter interacter) error {
 	return nil
 }
 
-// ServiceDescription describes a snappy systemd service
-type ServiceDescription struct {
-	SnapName        string
-	AppName         string
-	Version         string
-	Revision        int
-	Description     string
-	SnapPath        string
-	Start           string
-	Stop            string
-	PostStop        string
-	StopTimeout     time.Duration
-	Restart         systemd.RestartCondition
-	Type            string
-	AaProfile       string
-	BusName         string
-	UdevAppName     string
-	Socket          bool
-	SocketFileName  string
-	ListenStream    string
-	SocketMode      string
-	ServiceFileName string
-}
-
-func GenServiceFile(desc *ServiceDescription) string {
+func GenServiceFile(appInfo *snap.AppInfo) string {
 	serviceTemplate := `[Unit]
-Description={{.Description}}
-After=snapd.frameworks.target{{ if .Socket }} {{.SocketFileName}}{{end}}
-Requires=snapd.frameworks.target{{ if .Socket }} {{.SocketFileName}}{{end}}
+Description=service {{.App.Name}} for snap {{.App.Snap.Name}} - autogenerated DO NO EDIT
+After=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
+Requires=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
 X-Snappy=yes
 
 [Service]
-ExecStart=/usr/bin/ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.FullPathStart}}
+ExecStart=/usr/bin/ubuntu-core-launcher {{.App.SecurityTag}} {{.App.SecurityTag}} {{.FullPathStart}}
 Restart={{.Restart}}
 WorkingDirectory=/var{{.SnapPath}}
 Environment={{.EnvVars}}
-{{if .Stop}}ExecStop=/usr/bin/ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.FullPathStop}}{{end}}
-{{if .PostStop}}ExecStopPost=/usr/bin/ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.FullPathPostStop}}{{end}}
+{{if .App.Stop}}ExecStop=/usr/bin/ubuntu-core-launcher {{.App.SecurityTag}} {{.App.SecurityTag}} {{.FullPathStop}}{{end}}
+{{if .App.PostStop}}ExecStopPost=/usr/bin/ubuntu-core-launcher {{.App.SecurityTag}} {{.App.SecurityTag}} {{.FullPathPostStop}}{{end}}
 {{if .StopTimeout}}TimeoutStopSec={{.StopTimeout.Seconds}}{{end}}
-Type={{.Type}}
-{{if .BusName}}BusName={{.BusName}}{{end}}
+Type={{.App.Daemon}}
+{{if .App.BusName}}BusName={{.App.BusName}}{{end}}
 
 [Install]
 WantedBy={{.ServiceSystemdTarget}}
 `
 	var templateOut bytes.Buffer
 	t := template.Must(template.New("wrapper").Parse(serviceTemplate))
 
-	restartCond := desc.Restart.String()
+	restartCond := appInfo.RestartCond.String()
 	if restartCond == "" {
 		restartCond = systemd.RestartOnFailure.String()
 	}
+	socketFileName := ""
+	if appInfo.Socket {
+		socketFileName = filepath.Base(appInfo.ServiceSocketFile())
+	}
+	baseDir := appInfo.Snap.MountDir()
+	// this will remove the global base dir when generating the
+	// service file, this ensures that /snap/foo/1.0/bin/start
+	// is in the service file when the SetRoot() option
+	// is used
+	realBaseDir := stripGlobalRootDir(baseDir)
```

Comment:
given that u-d-f doesn't do this, this is mostly for the benefit of the tests wondering if there is some other way around that, anyway, may be for a later branch